File: spring-boot-java-web/src/main/java/com/example/webjsp/handlers/UUIDTypeHandler.java
package com.example.webjsp.handlers;

import org.apache.ibatis.type.*;
import java.sql.*;
import java.util.*;

@MappedJdbcTypes(JdbcType.OTHER)
@MappedTypes(UUID.class)
public class UUIDTypeHandler extends BaseTypeHandler<UUID> {

    @Override
    public void setNonNullParameter(PreparedStatement ps, int i, UUID parameter, JdbcType jdbcType) throws SQLException {
        ps.setObject(i, parameter); // no 3rd arg
    }

    @Override
    public UUID getNullableResult(ResultSet rs, String columnName) throws SQLException {
        String val = rs.getString(columnName);
        if (val == null)
            return null;
        return UUID.fromString(val);
    }

    @Override
    public UUID getNullableResult(ResultSet rs, int columnIndex) throws SQLException {
        // return rs.getObject(columnIndex, UUID.class);
        String val = rs.getString(columnIndex);
        if (val == null)
            return null;
        return UUID.fromString(val);
    }

    @Override
    public UUID getNullableResult(CallableStatement cs, int columnIndex) throws SQLException {
        //return cs.getObject(columnIndex, UUID.class);
        String val = cs.getString(columnIndex);
        if (val == null)
            return null;
        return UUID.fromString(val);
    }
}

File: spring-boot-kotlin-web/src/main/kotlin/com/example/webjsp/handlers/UUIDTypeHandler.kt
package com.example.webjsp.handlers

import org.apache.ibatis.type.BaseTypeHandler
import org.apache.ibatis.type.JdbcType
import org.apache.ibatis.type.MappedJdbcTypes
import org.apache.ibatis.type.MappedTypes
import java.sql.CallableStatement
import java.sql.PreparedStatement
import java.sql.ResultSet
import java.sql.SQLException
import java.util.*


@MappedJdbcTypes(JdbcType.OTHER)
@MappedTypes(UUID::class)
class UUIDTypeHandler: BaseTypeHandler<UUID>() {
    @Throws(SQLException::class)
    override fun setNonNullParameter(ps: PreparedStatement, i: Int, parameter: UUID?, jdbcType: JdbcType?) {
        ps.setObject(i, parameter) // no 3rd arg
    }

    @Throws(SQLException::class)
    override fun getNullableResult(rs: ResultSet, columnName: String?): UUID? {
        val src = rs.getString(columnName) ?: return null
        return UUID.fromString(src)
    }

    @Throws(SQLException::class)
    override fun getNullableResult(rs: ResultSet, columnIndex: Int): UUID? {
        val src = rs.getString(columnIndex) ?: return null
        return UUID.fromString(src)
    }

    @Throws(SQLException::class)
    override fun getNullableResult(cs: CallableStatement, columnIndex: Int): UUID? {
        val src = cs.getString(columnIndex) ?: return null
        return UUID.fromString(src)
    }
}